version: "3.8"

services:
  db.postgres132.devbox.local:
    image: postgres:13.2-alpine
    environment:
      POSTGRES_DB: dev
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: password
    volumes:
      - postgres132_db_data:/var/lib/postgresql/data
    networks:
      - postgres132_network_local
    ports:
      - 15432:5432
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 10s
      retries: 3
  pgadmin.postgres132.devbox.local:
    image: dpage/pgadmin4:5
    depends_on:
      - db.postgres132.devbox.local
    environment:
      PGADMIN_DEFAULT_EMAIL: pgadmin4@pgadmin.postgres132.devbox.local
      PGADMIN_DEFAULT_PASSWORD: password
    volumes:
      - ./config/servers.json:/pgadmin4/servers.json
      - ./config/pgpassfile:/pgadmin4/pgpassfile
      - postgres132_pgadmin_data:/var/lib/pgadmin
    networks:
      - postgres132_network_local
    ports:
      - 25432:80

volumes:
  postgres132_db_data:
  postgres132_pgadmin_data:

networks:
  postgres132_network_local:
    driver: bridge
